baron.jl

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env julia
+using JuMP
+using ArgParse
+
+using MINLPLib_jump
+using BARON
+
+
+include("common.jl")
+
+function parse_commandline()
+    s = ArgParseSettings()
+
+    @add_arg_table s begin
+        "--file", "-f"
+            help = "the minlplib data file (.jl)"
+            required = true
+        "--cpxlib", "-c"
+            help = "cplex lib name (must be included in path)"
+            arg_type = AbstractString
+        "--time-limit", "-t"
+            help = "puts a time limit on the sovler"
+            arg_type = Float64
+    end
+
+    return parse_args(s)
+end
+
+
+function main(parsed_args)
+
+    m = fetch_model(parsed_args["file"])
+    # this adds a model named m to the current scope
+
+    # By default, always run with CPLEX
+    parsed_args["cpxlib"] != nothing ? cpxlib = parsed_args["cpxlib"] : cpxlib = "libcplex1271.so"
```

Comment:
Does this still work if cplex is not installed?

Reply:
Yes. Baron will just say I cannot find the lib and switch back to Cbc.